v2/dtos/reading.go

```diff
@@ -47,6 +52,33 @@ type SimpleReading struct {
 	FloatEncoding string `json:"floatEncoding,omitempty"`
 }
 
+// Validate satisfies the Validator interface
+func (b BaseReading) Validate() error {
+	// check if the inner BinaryReading struct is empty
+	if !reflect.DeepEqual(b.BinaryReading, BinaryReading{}) {
```

Comment:
Can't we determine this based on ValueType, which should be common to both Binary & Simple, i.e should be in BaseReading. This will be an issue with consumers of the DTO's, i.e.App services, if we can't do this is a simple check of the ValueType.

Reply:
No, @lenny-intel , BinaryReading only contains mediaType, not ValueType.
If we do so, we need to move ValueType to BaseReading.  Is it ok?

Reply:
Yes, IMO, `ValueType` has to be in `BaseReading`. It is one of the key fields that App Services can filter one.

Reply:
Updated to define `ValueType` in `BaseReading` instead of `SimpleReading.`